qubes/storage/lvm.py

```diff
@@ -202,34 +256,77 @@ def __init__(self, volume_group, size=0, **kwargs):
 
         if self.snap_on_start or self.save_on_stop:
             self._vid_snap = self.vid + '-snap'
+        if self.save_on_stop:
+            self._vid_import = self.vid + '-import'
+
+        self._vid_current = None
 
         self._size = size
 
+    def _get_next_revision_number(self):
+        revisions = self._get_revisions('(|-back)')
+        if not revisions:
+            return 1
+        max_revision = max([
+            int(r.split('.')[0]) if '.' in r else 0
+            for r in revisions
+        ])
+        return max_revision + 1
+
+    def _refresh_vid_current(self):
+        revisions = self._get_revisions('')
+        if not revisions:
+            # fallback to old format - possibly pool
+            # not connected, don't crash caller
+            # of 'path' attribute access
+            self._vid_current = self.vid
+        else:
+            last_revision = \
+                max(revisions.items(), key=_revision_sort_key)[0]
+            self._vid_current = self.vid + '-' + last_revision
+
     @property
     def path(self):
-        return '/dev/' + self.vid
+        if self._vid_current is None:
+            self._refresh_vid_current()
+        return '/dev/' + self._vid_current
 
-    @property
-    def revisions(self):
+    def _get_revisions(self, suffix):
```

Comment:
> This is preserved - see that the most recent is excluded from self.revisions.

 I totally missed that there's now a wrapper function for the public interface. :+1: FWIW then.